### ISSUE TITLE
Debian package does not install cleanly on older ubuntu versions

### ISSUE THREAD

**Issue by abailly-iohk** [input-output-hk/mithril#834]
## Context & versions

Not sure which version of ubuntu but I would guess ubuntu 20.04 or something.

## Steps to reproduce

1. Download `.deb` package
2. Run `sudo dpkg -i mithril-client_0.2.14+b61945b_amd64.deb `

## Actual behavior

It fails with:

```
$ sudo dpkg -i mithril-client_0.2.14+b61945b_amd64.deb 
Selecting previously unselected package mithril-client.
(Reading database ... 31294 files and directories currently installed.)
Preparing to unpack mithril-client_0.2.14+b61945b_amd64.deb ...
Unpacking mithril-client (0.2.14+b61945b) ...
dpkg: dependency problems prevent configuration of mithril-client:
 mithril-client depends on libc6 (>= 2.34); however:
  Version of libc6:amd64 on system is 2.31-13+deb11u5.
 mithril-client depends on libssl3 (>= 3.0.0~~alpha1); however:
  Package libssl3 is not installed.

dpkg: error processing package mithril-client (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 mithril-client
```

## Expected behavior

IT should install cleanly or at least warn user about distribution version being too old/Provide some actionable feedback?

**Comment by abailly-iohk** [comment]
Ping @pgrange

**Comment by pgrange** [comment]
Thank you for the issue @abailly-iohk. Problem appears on Debian 11:

```
cat /etc/issue
Debian GNU/Linux 11 \n \l
```

**Comment by Alenar** [comment]
We merged #1043 that changes the ubuntu version used to build the packaged binaries from `ubuntu 22.04` to `ubuntu 20.04`.
That mean that we are now compatible with system bundling libc 2.31 or more (Debian 11+), instead of 2.35 minimum previously (Debian 12).

We also updated the doc to specify the minimum required libc version (ie: [run signer node guide](https://mithril.network/doc/next/manual/getting-started/run-signer-node#download-pre-built-binary)) and it's also specified in the github release notes as well (ie: [last unstable](https://github.com/input-output-hk/mithril/releases/tag/unstable)).

_Note: those changes are only available for unstable releases until the next (pre)release._

Does this solve this ticket for you @abailly-iohk, @pgrange ?
